libraries/AP_Scripting/examples/glide_into_wind.lua

```diff
@@ -0,0 +1,313 @@
+  local fs_long_actn = FS_LONG_ACTN:get()
+  
+  -- Check if glide_wind is not enabled or if fs_long_actn is not glide
+  if gw_enable == 1 and fs_long_actn ~= 2 then
+    send_to_gcs(_WARNING, 'GLIDE_WIND_ENABL is set, but FS_LONG_ACTN is not 2.')
+  end
+
+  -- All set, go to update
+  return update(), long_looptime
+end
+
+------------
+-- Main loop
+------------
+function update()
+  -- Check if state of GLIDE_WIND_WIND parameter changed, print every change
+  if gw_enable ~= GLIDE_WIND_ENABL:get() then
+    gw_enable = GLIDE_WIND_ENABL:get()
+    send_to_gcs(_INFO, 'LUA: GLIDE_WIND_ENABL: ' .. gw_enable)
+    -- If GLIDE_WIND_ENABL was enabled, warn if not FS_LONG_ACTN is set accordingly
+    if gw_enable == 1 then
+      local fs_long_actn = FS_LONG_ACTN:get()
+      if fs_long_actn ~=2 then
+        send_to_gcs(_WARNING, 'GLIDE_WIND_ENABL is set, but FS_LONG_ACTN is not 2.')
+      end
+    end
+  end
+
+  -- -- If feature is not enabled, loop slowly
+  if gw_enable == 0 then
+    return update, long_looptime
+  end
+  
+  -- GLIDE_WIND_ENABL is enabled, look for triggers
+  -- Monitor time since last gcs heartbeat
+  if last_seen == gcs:last_seen() then
+    link_lost_for = link_lost_for + looptime
+  else
+    -- There has been a new heartbeat, update last_seen and reset link_lost_for
+    last_seen = gcs:last_seen()
+    link_lost_for = 0
+  end
+
+  -- If link has been lost for more than FS_LONG_TIMEOUT and we are in FBWA, turn into wind
+  if link_lost_for > FS_LONG_TIMEOUT:get() * 1000 then
+    if FS_LONG_ACTN:get() == 2 then
+      if parse_flight_mode(vehicle:get_mode()) == "FBWA" then
+        -- Get the heading angle
+        hdg = math.floor(math.deg(ahrs:get_yaw()))
+        -- Get wind direction. Function wind_estimate returns x and y for direction wind blows in, add pi to get true wind dir
+        wind = ahrs:wind_estimate()
+        wind_dir_rad = math.atan(wind:y(), wind:x())+math.pi
+        wind_dir_180 = math.floor(wrap_180(math.deg(wind_dir_rad)))
+
+        -- P-regulator, calc error choose closes way - right or left.
+        error = wrap_180(wind_dir_180 - hdg)
+        
+        -- Multiply with kP and cast to int
+        roll = math.floor(error*GLIDE_WIND_RKP:get())
+        
+        -- Limit output
+        if roll > rlim then
+          roll = rlim
+        elseif roll < -rlim then
+          roll = -rlim
+        end
+
+        -- Check if we are close to target heading
+        if math.abs(error) < hdg_ok_lim then
+          -- If we have been close to target heading for hdg_ok_t_lim, stop overriding
+          if hdg_ok_t > hdg_ok_t_lim then
+            -- Reset roll input, send text to gcs
+            if override_enable then
+              RC_ROLL:set_override(1500)
+              send_to_gcs(_INFO, 'LUA: Gliding into wind, no more steering')
+            end
+            -- Do not override again until link has been recovered, set flag to false
```

Comment:
You are right, I shall explain that and put it in the top comment of the script. I will commit an updated version later today.
Except that I don't don't RC-override in general, there are 2-3 reasons:
1. If there is RC pilot in range, the pilot can control pitch and rudder but not roll as long as the channel is overridden. This is confusing and can be stressful and has the potential of locking up the mind of the pilot.
2. At touch down we want the roll attitude to be level. 
3. (I'm not 100% sure this is actually a problem/risk) Since the plane glides in FBWA with 0 throttle, it is likely close to stall speed after some time. Close to stall speed the roll response is smaller which could lead to bigger control error and hence bigger control outputs. During low speed this would further slow down the plane and also risk stalling the plane.

There is a problem in the current implementation:
If the RC pilot wants to cancel the ongoing glide into wind before the script says 'no more steering' and disables the flag override_enable, he/she flips flight mode and gets full control, ok. Later when the RC-pilot returns to FBWA, and the GCS link has still not recovered, the script again will try to steer into wind.
I will present an update version later today where this is adressed.